components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -0,0 +1,202 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCBannerView.h"
+
+#import "MDCBannerParams.h"
+#import "MDCBannerViewLayout.h"
+#import "MaterialColorScheme.h"
+#import "MaterialTypography.h"
+
+@interface MDCBannerView () <MDCBannerViewLayoutDataSource>
+
+@property(nonatomic, readwrite, weak) UILabel *textLabel;
+@property(nonatomic, readwrite, weak) UIImageView *iconImageView;
+@property(nonatomic, readwrite, strong) NSMutableArray<__kindof UIButton *> *buttons;
+
+@property(nonatomic, readwrite, strong) MDCBannerViewLayout *layout;
+
+@end
+
+@implementation MDCBannerView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonBannerViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonBannerViewInit];
+  }
+  return self;
+}
+
+- (void)commonBannerViewInit {
+  self.backgroundColor = [UIColor whiteColor];
+  _buttons = [[NSMutableArray alloc] init];
+}
+
+#pragma mark - Property Setter and Getter
+
+- (void)setText:(NSString *)text {
+  if (text) {
+    _text = text;
+    UILabel *textLabel = self.textLabel;
+    if (!textLabel) {
+      textLabel = [[UILabel alloc] init];
+      textLabel.font = [MDCTypography body2Font];
+      MDCSemanticColorScheme *colorScheme = [[MDCSemanticColorScheme alloc] init];
+      textLabel.textColor = colorScheme.onSurfaceColor;
+      textLabel.numberOfLines = 3;
+      textLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+      self.textLabel = textLabel;
+      [self addSubview:textLabel];
+    }
+    textLabel.text = text;
+  } else {
+    _text = nil;
+    [self.textLabel removeFromSuperview];
+  }
+  [self.layout reloadData];
+  [self setNeedsLayout];
+}
+
+- (void)setImage:(UIImage *)image {
+  if (image) {
+    _image = image;
+    if (!self.iconImageView) {
+      UIImageView *iconImageView = [[UIImageView alloc] initWithImage:image];
+      iconImageView.frame = CGRectMake(0, 0, kIconImageSideLength, kIconImageSideLength);
```

Comment:
40.0 is not a standard size for our icons. on IOS our icon sizes are 24, 36, 48, etc.
I would also consider letting the image size be determined by given image instead of forcing a single size. That would also work better with Dynamic Types, in case we decide to change the size of the image when the text size changes (which we're not yet sure about, btw).

Reply:
Mark this one for discussion.

Reply:
After discussion, keeping it as fixed is acceptable. Will need to confirm with designers, and probably expose a method to change the image view frame.